packages/headless/src/controllers/commerce/core/sub-controller/headless-sub-controller.ts

```diff
@@ -0,0 +1,28 @@
+import {CommerceEngine} from '../../../../app/commerce-engine/commerce-engine';
+import {
+  buildCoreInteractiveResult,
+  InteractiveResult,
+  InteractiveResultProps,
+} from '../result-list/headless-core-interactive-result';
+
+export interface SolutionTypeSubControllers {
+  interactiveResult: (props: InteractiveResultProps) => InteractiveResult;
+}
+
+interface SubControllerProps {
+  responseIdSelector: () => string;
+}
+
+export function buildSolutionTypeSubControllers(
```

Comment:
Having this helper method allows us to add sub-controller support for solution type controllers without changing them individually. When we'll add support for pagination, sort and facets as sub-controllers in [CAPI-719](https://coveord.atlassian.net/browse/CAPI-719), this is where we'll do changes.